### PR TITLE
Update WMI adapter to perform query based on provided properties

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

Previously, all the properties for a WMI class gets returned.  With this change, the specified properties get turned into a query.  This can actually improve performance if the WMI provider supports it, but otherwise it also means the output won't have information that the configuration did not request.

If a property has a value, then that value is used as part of the query.

Example output using the updated wmi inventory example:

```yaml
results:
- metadata:
    Microsoft.DSC:
      duration: PT1.0536124S
  name: WMI
  type: Microsoft.Windows/WMI
  result:
    actualState:
    - Model: Virtual Machine
      Domain: WORKGROUP
      Name: CPC-slee-8MW0NA
      Manufacturer: Microsoft Corporation
      TotalPhysicalMemory: 68665831424
    - Caption: Microsoft Windows 11 Enterprise
      OSArchitecture: 64-bit
      OSLanguage: 1033
      Version: 10.0.22631
    - SerialNumber: 9040-0535-4797-8220-8672-1612-79
      Manufacturer: Microsoft Corporation
      Model: null
    - SerialNumber: 0000-0011-2510-4748-5887-4988-97
      Manufacturer: Microsoft Corporation
      Version: VRTUAL - 1
    - MACAddress: 00:22:48:C1:70:B2
      ServiceName: netvsc
      NetConnectionID: Ethernet 6
      NetConnectionStatus: 2
      Name: 'Microsoft Hyper-V Network Adapter #5'
      AdapterType: Ethernet 802.3
```

Also fixed how the adapter writes traces so they show up correctly.

## PR Context

Address a part of https://github.com/PowerShell/DSC/issues/475

### CODE REVIEW

File: wmi-adapter/wmi.resource.ps1 [hunk 1]
```diff
@@ -9,10 +9,28 @@ param(
     $stdinput
 )
 
+trap {
```
Comment (anmenaga) [inline wmi-adapter/wmi.resource.ps1:12]: Would be good to have a comment about what this `trap` does.

File: dsc/examples/wmi_inventory.dsc.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          serialnumber:
+      - name: network adapter
+        type: root.cimv2/Win32_NetworkAdapter
+        properties:
+          name:
+          macaddress:
+          adaptertype:
+          netconnectionid:
+          serviceName:
+          netconnectionstatus: 2
```
Comment (anmenaga) [inline dsc/examples/wmi_inventory.dsc.yaml:42]: Is my understanding correct that this is the only instance in this file that will use `WHERE` clause?

**Review by anmenaga — APPROVED** [review]

File: dsc/examples/wmi_inventory.dsc.yaml (hunk 2 above)
Comment (SteveL-MSFT) [reply dsc/examples/wmi_inventory.dsc.yaml:42]: Yes

File: wmi-adapter/wmi.resource.ps1 (hunk 1 above)
Comment (SteveL-MSFT) [reply wmi-adapter/wmi.resource.ps1:12]: ok, will add comment

**Pull request merged**